### DISCUSSION TITLE
Use custom OpenAPI Schema with kustomize

### DISCUSSION THREAD

**Discussion by stoetti** [fluxcd/flux2#1880] [Q&A]
We have our own CRD for deploying our service in k8s and are moving to GitOps principle for deployment. In order to use kustomize at its best we want to use strategic merge for our CRD which makes necessary to provide a custom OpenAPI schema as described here https://github.com/kubernetes-sigs/kustomize/blob/master/examples/customOpenAPIschema.md

Locally executing kustomize build it works as expected. When applied using flux in the cluster it seems that the custom schema is not taken into account. Should that be supported by flux?

**Comment by stefanprodan** [comment]
Flux uses the same Go package as the kustomize CLI, so yes it should work the same as running `kustomize build` assuming you use the same version as Flux https://github.com/fluxcd/kustomize-controller/blob/main/go.mod#L38

Please open an issue in [kustomize-controller](https://github.com/fluxcd/kustomize-controller) and provide manifests to replicate the bug.

**Reply by stoetti** [reply]
created the issue https://github.com/fluxcd/kustomize-controller/issues/436
